### PR TITLE
Some Tweaks To #7

### PR DESCRIPTION
**Pull request by calmilamsy**
The main changes are:

- Change project from tabs to spaces, in line with IDE defaults.
- Removed a leftover check from the buildscript.
- Added spaces after commas and ifs. Why, Martin?
- Renamed some methods to make more sense. Why are they named internal if they have no default impl?
  - `changeEnergy` to `adjustEnergy`
- Added some documentation where it was glaringly missing.
- Removed energy network type as it's entirely unused.
- Renamed `base` package to `templates` to be in line with what StAPI does.

### CODE REVIEW

File: src/main/java/net/teamterminus/machineessentials/network/NetworkWalker.java [hunk 1]
```diff
@@ -14,199 +14,199 @@
 /**
  * Travels across a network recording possible paths between its components.
  * <p>
- * This class uses <code>NetworkWire</code> to differentiate between the medium (wires) and endpoints (devices) of a network to build paths.
- * @param <T> Any type that extends <code>NetworkComponent</code>
+ * This class uses <code>INetworkWire</code> to differentiate between the medium (wires) and endpoints (devices) of a network to build paths.
```
Comment (MartinSVK12) [inline src/main/java/net/teamterminus/machineessentials/network/NetworkWalker.java:17]: How did the "I" get here again

**Review by MartinSVK12 — CHANGES_REQUESTED** [review]
Just one small regression in `NetworkWalker`s javadoc, otherwise fine i guess?

**Comment by MartinSVK12** [comment]
Okay, fixed that.

**Review by MartinSVK12 — APPROVED** [review]

File: src/main/java/net/teamterminus/machineessentials/energy/electric/template/ElectricDeviceBlockEntity.java [hunk 2]
```diff
@@ -21,49 +21,49 @@ public void tick() {
...
+            if (be instanceof ElectricWire){
+                receiveEnergy(dir, getMaxInputAmperage());
             }
         }
     }
 
     @Override
     public long receiveEnergy(@NotNull Direction dir, long amperage) {
-        if(amperage > getMaxInputAmperage()){
+        if (amperage > getMaxInputAmperage()){
```
Comment (MrMasrozYTLIVE) [inline src/main/java/net/teamterminus/machineessentials/energy/electric/template/ElectricDeviceBlockEntity.java:38]: If we are talking about putting spaces everywhere then we also should put space here after `if ()`

File: src/main/java/net/teamterminus/machineessentials/energy/electric/template/ElectricDeviceBlockEntity.java [hunk 3]
```diff
@@ -21,49 +21,49 @@ public void tick() {
...
         long willUseAmps = 0;
         BlockEntity blockEntity = MachineEssentials.getBlockEntity(dir, world, this);
-        if(blockEntity instanceof ElectricWire wire) {
+        if (blockEntity instanceof ElectricWire wire) {
             //for every known path
             for (NetworkPath path : energyNet.getPathData(wire.getPosition())) {
                 long pathLoss = 0;
                 //ignore itself or non-electric components in the path
-                if(path.target == this || !(path.target instanceof Electric dest)){
+                if (path.target == this || !(path.target instanceof Electric dest)){
```
Comment (MrMasrozYTLIVE) [inline src/main/java/net/teamterminus/machineessentials/energy/electric/template/ElectricDeviceBlockEntity.java:49]: Also space here after `if ()`

File: src/main/java/net/teamterminus/machineessentials/energy/electric/template/ElectricDeviceBlockEntity.java [hunk 4]
```diff
@@ -21,49 +21,49 @@ public void tick() {
...
+                if (dest.canProvide(path.targetDirection.getOpposite())) {
                     if (canReceive(dir)) {
                         //get max voltage from destination
                         //limit amps to maximum available from destination
                         long voltage = dest.getMaxOutputVoltage();
                         amperage = Math.min(amperage, (dest.getMaxOutputAmperage() - dest.getAmpsCurrentlyUsed()));
                         //calculate path loss
                         for (NetworkComponent component : path.path) {
-                            if(component instanceof ElectricWire pathWire){
+                            if (component instanceof ElectricWire pathWire){
```
Comment (MrMasrozYTLIVE) [inline src/main/java/net/teamterminus/machineessentials/energy/electric/template/ElectricDeviceBlockEntity.java:62]: And also space here after `if ()`

File: src/main/java/net/teamterminus/machineessentials/energy/electric/template/ElectricDeviceBlockEntity.java [hunk 5]
```diff
@@ -21,49 +21,49 @@ public void tick() {
...
                                 pathLoss += pathWire.getProperties().material().lossPerBlock();
                             }
                         }
-                        if(pathLoss >= voltage){
+                        if (pathLoss >= voltage){
```
Comment (MrMasrozYTLIVE) [inline src/main/java/net/teamterminus/machineessentials/energy/electric/template/ElectricDeviceBlockEntity.java:66]: Aaaannd here

File: src/main/java/net/teamterminus/machineessentials/energy/electric/template/ElectricDeviceBlockEntity.java [hunk 6]
```diff
@@ -72,29 +72,29 @@ public long receiveEnergy(@NotNull Direction dir, long amperage) {
                         boolean pathBroken = false;
                         //handle wires with insufficient voltage rating
                         for (NetworkComponent pathBlockEntity : path.path) {
-                            if(pathBlockEntity instanceof ElectricWire pathWire){
-                                if(pathWire.getVoltageRating() < voltage){
+                            if (pathBlockEntity instanceof ElectricWire pathWire){
```
Comment (MrMasrozYTLIVE) [inline src/main/java/net/teamterminus/machineessentials/energy/electric/template/ElectricDeviceBlockEntity.java:75]: As well as these  2 IFs on 75 and 76

File: src/main/java/net/teamterminus/machineessentials/energy/electric/template/ElectricDeviceBlockEntity.java [hunk 7]
```diff
@@ -72,29 +72,29 @@ public long receiveEnergy(@NotNull Direction dir, long amperage) {
...
                                     pathBroken = true;
                                     pathVoltage = Math.min(pathWire.getVoltageRating(), pathVoltage);
                                     break;
                                 }
                             }
                         }
-                        if(pathBroken) continue;
+                        if (pathBroken) continue;
 
-                        if(pathVoltage > 0){
```
Comment (MrMasrozYTLIVE) [inline src/main/java/net/teamterminus/machineessentials/energy/electric/template/ElectricDeviceBlockEntity.java:86]: Maybe also here?xD

File: src/main/java/net/teamterminus/machineessentials/energy/electric/template/ElectricDeviceBlockEntity.java [hunk 8]
```diff
@@ -72,29 +72,29 @@ public long receiveEnergy(@NotNull Direction dir, long amperage) {
...
+                        if (pathVoltage > 0){
                             //handle device over-voltage
-                            if(pathVoltage > getMaxInputVoltage()){
+                            if (pathVoltage > getMaxInputVoltage()){
```
Comment (MrMasrozYTLIVE) [inline src/main/java/net/teamterminus/machineessentials/energy/electric/template/ElectricDeviceBlockEntity.java:88]: I love that there's so many of them. Gotta have so much fun writing these comments

File: src/main/java/net/teamterminus/machineessentials/network/NetworkWalker.java [hunk 9]
```diff
@@ -19,194 +19,194 @@
...
+
+    public NetworkWalker(World world, Vec3i source, int walkedBlocks, List<NetworkPath> routes){
+        this.world = world;
+        this.walkedBlocks = walkedBlocks;
+        this.currentPos = source;
+        this.root = this;
+        this.routes = routes;
+    }
+
+    public static <T extends NetworkComponent> List<NetworkPath> createNetworkPaths(World world, Vec3i source){
```
Comment (MrMasrozYTLIVE) [inline src/main/java/net/teamterminus/machineessentials/network/NetworkWalker.java:49]: Missing some spaces after `if ()` here

File: src/main/java/net/teamterminus/machineessentials/energy/electric/template/ElectricDeviceBlockEntity.java [hunk 10]
```diff
@@ -21,49 +21,49 @@ public void tick() {
...
-            if(be instanceof ElectricWire){
-                receiveEnergy(dir,getMaxInputAmperage());
+            if (be instanceof ElectricWire) {
+                receiveEnergy(dir, getMaxInputAmperage());
             }
         }
     }
 
     @Override
     public long receiveEnergy(@NotNull Direction dir, long amperage) {
```
Comment (MrMasrozYTLIVE) [inline src/main/java/net/teamterminus/machineessentials/energy/electric/template/ElectricDeviceBlockEntity.java:37]: Oh my god why there's so many IFs? This is such big nest of IFs

Comment (MrMasrozYTLIVE) [reply src/main/java/net/teamterminus/machineessentials/energy/electric/template/ElectricDeviceBlockEntity.java:37]: In this method I mean. Just look at the end of the method there's at least like 10 closing brackets before `return` and the end of the method

Comment (MrMasrozYTLIVE) [reply src/main/java/net/teamterminus/machineessentials/energy/electric/template/ElectricDeviceBlockEntity.java:37]: > @MartinSVK12 's problem.

I've fixed it and could've push but right now I am not at my pc so ehhh

**Pull request merged**